Devscord.DiscordFramework/Middlewares/Factories/ReactionContextFactory.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using Discord;
+using Discord.WebSocket;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+
+namespace Devscord.DiscordFramework.Middlewares.Factories
+{
+    internal class ReactionContextFactory : IContextFactory<Tuple<SocketReaction, IUserMessage>, ReactionContext>
+    {
+        private readonly ChannelContextFactory _channelContextFactory;
+        private readonly UserContextsFactory _userContextsFactory;
+        private readonly MessageContextFactory _messageContextFactory;
+
+        public ReactionContextFactory()
+        {
+            this._channelContextFactory = new ChannelContextFactory();
+            this._userContextsFactory = new UserContextsFactory();
+            this._messageContextFactory = new MessageContextFactory();
+        }
+
+        public ReactionContext Create(Tuple<SocketReaction, IUserMessage> reactionAndUserMessage)
```

Comment:
staraj sie unikać tuple
a jak już musisz to rób to z nazywaniem propertiesów (dodane w C# 7)